sklego/meta/ordinal_classification.py

```diff
@@ -0,0 +1,185 @@
+    by Eibe Frank and Mark Hall.
+
+    """
+
+    def __init__(self, estimator, *, n_jobs=None, use_calibration=True):
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+        self.use_calibration = use_calibration
+
+    def fit(self, X, y):
+        """Fit the `OrdinalClassifier` model on training data `X` and `y` by fitting its underlying estimators on
+        N-1 datasets `X` and `y` for each class `y_label` in `y` except `y.max()`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features )
+            The training data.
+        y : array-like of shape (n_samples,)
+            The target values.
+
+        Returns
+        -------
+        self : OrdinalClassifier
+            Fitted model.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not a classifier or if it does not implement `.predict_proba()`.
+        """
+
+        if not is_classifier(self.estimator):
+            raise ValueError("The estimator must be a classifier.")
+
+        if not hasattr(self.estimator, "predict_proba"):
+            raise ValueError("The estimator must implement `.predict_proba()` method.")
+
+        X, y = check_X_y(X, y, estimator=self)
+
+        self.classes_ = np.sort(np.unique(y))
+        self.n_features_in_ = X.shape[1]
+
+        if self.n_jobs is None or self.n_jobs == 1:
+            self.estimators_ = {y_label: self._fit_binary_estimator(X, y, y_label) for y_label in self.classes_[:-1]}
+        else:
+            self.estimators_ = dict(
+                zip(
+                    self.classes_[:-1],
+                    Parallel(n_jobs=self.n_jobs)(
+                        delayed(self._fit_binary_estimator)(X, y, y_label) for y_label in self.classes_[:-1]
+                    ),
+                )
+            )
+        return self
+
+    def predict_proba(self, X):
+        """Predict class probabilities for samples in `X`. The class probabilities of a sample are computed as the
+        difference between the probability of the sample to be in class `y_label` and the probability of the sample to
+        be in class `y_label - 1`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data to predict.
+
+        Returns
+        -------
+        array-like of shape (n_samples, n_classes)
+            The predicted class probabilities.
+
+        Raises
+        ------
+        ValueError
+            If `X` has a different number of features than the one seen during `fit`.
+        """
+        check_is_fitted(self, ["estimators_", "classes_"])
+
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError(f"X has {X.shape[1]} features, expected {self.n_features_in_} features.")
+
+        raw_proba = np.array([estimator.predict_proba(X)[:, 1] for estimator in self.estimators_.values()]).T
```

Comment:
Some food for thought: I also wonder if we want to expose a method that returns these probabilities directly

Reply:
Makes sense to me!